Source/LuaBridge/RefCountedPtr.h

```diff
@@ -66,13 +66,13 @@ class RefCountedPtr : private detail::RefCountedPtrBase
 
         @param p The optional, existing pointer to assign from.
     */
-    RefCountedPtr(T* p = 0) : m_p(p) { ++getRefCounts()[m_p]; }
+    RefCountedPtr(T* const p = nullptr) : m_p(nullptr) { *this = p; }
```

Comment:
I'd actually prefer to have constructor/destructor logic separated not gathered in a single function.
Having the logic separated we could use the swap idiom:
```C++
RefCountedPtr<T>& operator=(T* const p)
{
    RefCountedPtr<T> tmp(p);
    std::swap(m_p, tmp.m_p);
    return *this;
}
```

Reply:
I decided to do it in this way to keep the logic at one place.
Otherwise the solved issues have to be tackled at multiple places:
- NULL pointer issue: three times in the constructors and once in use_count
- Nested objects issue: two times in the assign operators

Reply:
It seems like the swap idiom could be solved with a separate move ctor.

Reply:
Ok, now I got it. This sounds reasonable and makes the source even more easier to understand.
I'm going to change this during the weekend.